cli/lsp/tsc.rs

```diff
@@ -0,0 +1,1061 @@
+// Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
+
+use super::analysis::ResolvedImport;
+use super::state::ServerStateSnapshot;
+use super::text;
+
+use crate::media_type::MediaType;
+use crate::tsc::ResolveArgs;
+use crate::tsc_config::TsConfig;
+
+use deno_core::error::custom_error;
+use deno_core::error::AnyError;
+use deno_core::json_op_sync;
+use deno_core::serde::Deserialize;
+use deno_core::serde_json;
+use deno_core::serde_json::json;
+use deno_core::serde_json::Value;
+use deno_core::JsRuntime;
+use deno_core::ModuleSpecifier;
+use deno_core::OpFn;
+use deno_core::RuntimeOptions;
+use deno_core::Snapshot;
+use regex::Captures;
+use regex::Regex;
+use std::borrow::Cow;
+use std::collections::HashMap;
+
+fn display_parts_to_string(
+  maybe_parts: Option<Vec<SymbolDisplayPart>>,
+) -> Option<String> {
+  if let Some(parts) = maybe_parts {
+    Some(
+      parts
+        .into_iter()
+        .map(|p| p.text)
+        .collect::<Vec<String>>()
+        .join(""),
+    )
+  } else {
+    None
+  }
```

Comment:
Nit: instead of this `if let ... else` you can use `Option::map`

```rust
maybe_parts.map(|parts| {
  parts.into_iter().map(|p| p.text).collect::<Vec<String>>().join("")
})

```